include/deal.II/numerics/matrix_creator.templates.h

```diff
@@ -632,22 +632,56 @@ namespace MatrixCreator
 
 
     } // namespace AssemblerBoundary
+
+    
+
+    template <typename T>
+    using compress_t = decltype(std::declval<const T>().compress(
+      std::declval<	VectorOperation>()));
+
+
+
+    template <typename T>
+    constexpr bool has_compress = dealii::internal::is_supported_operation<compress_t, T>;
+
+
+
+    template <typename MatrixType,
+              std::enable_if_t<has_compress<MatrixType>, MatrixType>
+                * = nullptr>
+    void
+    compress(const MatrixType &A, const VectorOperation::values operation)
+    {
+      A.compress(operation);
+    }
+
+
+
+    template <typename MatrixType,
+              std::enable_if_t<!has_compress<MatrixType>, MatrixType>
+                * = nullptr>
+    void
+    compress(const MatrixType &, const VectorOperation::values )
+    {
+      // nothing to do
+    }
```

Comment:
This is the relevant part. `FullMatrix` objects do not have `compress()` functions.

Reply:
This works, and also supports matrix-like types we don't control. The alternative would have been to add `compress()` to `FullMatrix`, but this is more general. Either approach would have worked for me.